django/db/backends/base/schema.py

```diff
@@ -1077,6 +1077,11 @@ def _unique_sql(self, model, fields, name, condition=None):
             'constraint': constraint,
         }
 
+    def _get_index_condition_sql(self, condition):
+        if not condition or not self.connection.features.supports_partial_indexes:
+            return ''
+        return ' WHERE ' + condition
```

Comment:
`_get_index_condition_sql()` in the current form don't work properly because it hides database exceptions when we try to create partial indexes on databases which don't support them, it will create a normal index without a condition. This is not an expected behavior. We could create `_index_condition_sql()` without checking `self.connection.features.supports_partial_indexes`, e.g.:
```python
    def _index_condition_sql(self, condition):
        if condition:
            return ' WHERE ' + condition
        return ''
```
but TBH I don't see much value in it.

Reply:
I have the same comment to the `_get_index_include_sql()`, this shouldn't check `supports_include_indexes` feature, IMO.

Reply:
OK, it makes sense to add a simple `_index_condition_sql()` if we take into account this PR:
```python
def _index_condition_sql(self, condition):
    if condition:
        return ' WHERE ' + condition
    return ''
```

Reply:
I agree that it's better to crash with a database error if one tries to add an index where these features isn't supported. 

I'll remove these checks.

Reply:
I've removed the `self.connection.features.supports_partial_indexes` and `self.connection.features.supports_include_indexes`. Some testcases needed to be updated as they where built expecting the silent fallback on non-supported features.

Reply:
And idea might be to add a check for feature support on `Index` and `UniqueConstraint` similar to how `PostgresIndex.check_supported` is implemented.

Reply:
> I've removed the `self.connection.features.supports_partial_indexes` and `self.connection.features.supports_include_indexes` checks. Some testcases needed to be updated as they where built expecting the silent fallback on non-supported features.

We added system checks for partial indexes support (see 53d229ff632c2a3e547f2820a94239f38ba4d4ac), so we should restore the previous behavior.

IMO we can ignore indexes with `INCLUDE` if they're not support and add a similar checks for them (check can be added in a separate PR/commit).

I moved `BaseDatabaseSchemaEditor._index_condition_sql()` to #12787.

Reply:
I've rebased the branch and removed the commit that was added in #12787. The test failure seems like an unrelated flaky test.